.github/workflows/build-and-push-python-package-test.yml

```diff
@@ -0,0 +1,39 @@
+name: Build Python Package
+
+on:
+  pull_request:
+  push:
+    branches:
+    - master
+
+jobs:
+  python:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.9']
+    defaults:
+      run:
+        working-directory: ./anubis-management-api
+    steps:
+      - uses: actions/checkout@v1
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: 'pipenv'
+          cache-dependency-path: |
+            anubis-management-api/Pipfile.lock
+      - name: Install pipenv
+        run: pip install pipenv
+      - name: Build distribution
+        run: |
+          pipenv run python setup.py bdist_wheel
+      - name: Publish package
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
+          password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+          repository_url: https://test.pypi.org/legacy/
```

Comment:
that's seem to be the only difference, so i would reccomend to have a test to set it, rather than duplicating the whole script

Reply:
The other difference is that this is `on release` whilst the other is `on pull request and merge to master`. I'm not sure you can conditionally branch events without writing two jobs.

Reply:
@chicco785 Wouldn't the top of the file's condition for the job:

```
on:
  release:
    types: [published]
```

and the conditionals in the job itself be kind of redundant? Since that statement above says that the job triggers when a published release happens anyways. Or am I getting this wrong?

Reply:
sorry, i didn't paste in the part related to push and pr, have a look